otter/convergence/model.py

```diff
@@ -112,6 +117,14 @@ class StepResult(Names):
     """
 
 
+@sumtype
+class ErrorReason(object):
+    """A reason for a step to be in a RETRY or FAILURE state."""
+    Exception = constructor('exc_info')
+    Other = constructor(reason=attr.ib(validator=instance_of((unicode, str))))
```

Comment:
How about `String = constructor...`? That way it will be instantiated by `Exception.String` which I think makes more sense than `Exception.Other`?